### PR TITLE
New: Add `vue/no-deprecated-v-on-number-modifiers` rule

### PR DESCRIPTION
**Pull request by yoyo930021**
This PR adds the `vue/no-deprecated-v-on-number-modifiers` rule.

This rule reports use of deprecated `KeyboardEvent.keyCode` modifier on `v-on` directive

ref #1035

### CODE REVIEW

File: docs/rules/no-deprecated-v-on-number-modifiers.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+This rule reports use of deprecated `KeyboardEvent.keyCode` modifier on `v-on` directive (in Vue.js 3.0.0+)
+
+<eslint-code-block fix :rules="{'vue/no-deprecated-v-on-number-modifiers': ['error']}">
+
+```vue
+<template>
+  <!-- ✓ GOOD -->
+  <input v-on:keyup.page-down="onArrowUp">
+  <input @keyup.page-down="onArrowUp">
+  <input @keyup.9="onArrowUp"> <!-- 9 is KeyboardEvent.key -->
```
Comment (ota-meshi) [inline docs/rules/no-deprecated-v-on-number-modifiers.md]: Thank you for this PR.
Almost look good to me.

I could not set KeyboardEvent.key to 9. Can you tell me which browser can reproduce it?

Comment (yoyo930021) [reply docs/rules/no-deprecated-v-on-number-modifiers.md]: When you keyup `9` on 60% keyborad.

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values
At the bottom of the linked page.

**Comment by yoyo930021** [comment]
@ota-meshi 
Screen Shot 2020-03-19 at 5 29 17 PM

File: docs/rules/no-deprecated-v-on-number-modifiers.md (hunk 1 above)
Comment (ota-meshi) [reply docs/rules/no-deprecated-v-on-number-modifiers.md]: Thank you for teaching me! NUMPAD! I misunderstood the Tab key.

**Pull request merged**